modules/dnn/src/layers/gatherND.cpp

```diff
@@ -0,0 +1,176 @@
+
+        std::vector<Mat> inputs, outputs;
+        inputs_arr.getMatVector(inputs);
+        outputs_arr.getMatVector(outputs);
+
+        const Mat& data = inputs[0];
+        const Mat& indices = inputs[1];
+        Mat& out = outputs[0];
+
+        int dtype = data.depth();
+        int itype = indices.depth();
+
+        switch (itype) {
+            case CV_32S:
+            {
+                switch (dtype) {
+                    case CV_8U: forward_impl<int32_t, uchar>(data, indices, out); break;
+                    case CV_8S: forward_impl<int32_t, schar>(data, indices, out); break;
+                    case CV_32S: forward_impl<int32_t, int32_t>(data, indices, out); break;
+                    case CV_16F: forward_impl<int32_t, float16_t>(data, indices, out); break;
+                    case CV_32F: forward_impl<int32_t, float>(data, indices, out); break;
+                    case CV_64F: forward_impl<int32_t, double>(data, indices, out); break;
+                    default: CV_Error(Error::StsNotImplemented, "Unsupported data type");
+                }
+            }
+            case CV_64S:
+            {
+                switch (dtype) {
+                    case CV_8U: forward_impl<int64_t, uchar>(data, indices, out); break;
+                    case CV_8S: forward_impl<int64_t, schar>(data, indices, out); break;
+                    case CV_32S: forward_impl<int64_t, int32_t>(data, indices, out); break;
+                    case CV_16F: forward_impl<int64_t, float16_t>(data, indices, out); break;
+                    case CV_32F: forward_impl<int64_t, float>(data, indices, out); break;
+                    case CV_64F: forward_impl<int64_t, double>(data, indices, out); break;
+                    default: CV_Error(Error::StsNotImplemented, "Unsupported data type");
+                }
+            }
+        }
+
+    }
+
+    template <typename iT, typename dT>
+    void forward_impl(const Mat& data, const Mat& indices, Mat& out)
+    {
+        CV_Assert(out.isContinuous());
+        const iT* indices_ptr = indices.ptr<iT>();
+        const dT* data_ptr = data.ptr<dT>();
+        dT* out_ptr = out.ptr<dT>();
+
+        int r = data.dims;
+        int q = indices.dims;
+        int last_indices_dim = indices.size[q - 1];
+
+        std::vector<int> data_strides(r);
+        data_strides[r - 1] = 1;
+        for (int i = r - 2; i >= 0; --i)
+            data_strides[i] = data_strides[i + 1] * data.size[i + 1];
+
+        std::vector<int> indices_strides(q);
+        indices_strides[q - 1] = 1;
+        for (int i = q - 2; i >= 0; --i)
+            indices_strides[i] = indices_strides[i + 1] * indices.size[i + 1];
+
+        const int outer_size = indices.total() / last_indices_dim;
+        const int inner_size = out.total() / outer_size;
+
+        for (size_t i = 0; i < outer_size; ++i)
+        {
+            std::vector<int> sliced_indices(indices_ptr + i * last_indices_dim, indices_ptr + (i + 1) * last_indices_dim);
```

Comment:
you do not need vector here. just:
```
int* sliced_indices = indices_ptr + i * last_indices_dim;
```

Reply:
This leads to crush during build time

Reply:
I just pushed fix. `indices_ptr` has template type `iT*`  and `sliced_indices` should. It's not always `int`. The solution with vector just hides the problem.